### ISSUE TITLE
Static homepage and '/blog' gatsby blog with netlify cms

### ISSUE THREAD

**Issue by AdvaithD** [gatsbyjs/gatsby#6037]
Hey guys I'm using the following template: https://gatsby-netlify-cms.netlify.com/ and wanted to host this at the '/blog' path along with a static bootstrap landing page. How do I go about this?

**Comment by Khaledgarbaya** [comment]
Hey @AdvaithD, you can do that by renaming `pages/index.js` to `pages/blog.js`.

**Comment by AdvaithD** [comment]
Hey @Khaledgarbaya dont exactly get it but I was looking for some help in having a static index.html along with assets( /css /js etc) and the above template for '/blog'

I've implemented custom paths but having some difficulty figuring out hwo to have all of this in one repo.

Where would I put in the static html and assets for the homepage?

**Comment by Khaledgarbaya** [comment]
Hey @AdvaithD I highly recommend that you go through the [tutorials](https://www.gatsbyjs.org/tutorial/) to get familiar with Gatsby.

TLDR; version.
1. You can have your entire website in one repo.

1. There are 2 types of routes, static when you know all the part that will define your route like /home. And dynamic when part of your route is only know at runtime like blog/:slug. so if you want to have pages like `mywebsite.com/blog` and `mywebsite.com/blog` you need to create javascript files in the pages directory.

your project structure will look like this 

```
├── LICENSE
├── README.md
├── gatsby-config.js
├── gatsby-node.js
├── node_modules
├── package-lock.json
├── package.json
├── src
│   ├── layouts
│   ├── pages
│   │    ├──  index.js
│   │    ├──  blog.js
│   │    ├──  contact.js
│   └── templates
└── static
```
This *.js files will have [react component](https://reactjs.org/docs/react-component.html) inside of them. When you build your website Gatsby will take care of creating the pages and put them in `public` folder and to host the website you just need to deploy the content of that folder.

1. If you have your code hosted on Github feel free share it and I can help you when possible

**Comment by AdvaithD** [comment]
@Khaledgarbaya thanks a lot, I am aware of the react component structure and the dynamic routes stuff, although when I change the index.js to blog.js and build with --prefix-paths I get this on '/'

![](https://i.gyazo.com/2edbe2d593024ec32393cc5cf9db26da.png)

**Comment by AdvaithD** [comment]
@Khaledgarbaya I just messed around and ran `gatsby develop` and the homepage shows up, however, when I go to '/blog' it still redirects to the homepage.

Thanks a lot for that TLDR btw, solved lots of my ambiguity!

**Comment by Khaledgarbaya** [comment]
@AdvaithD yeah the `/` path needs an `index.js` file otherwise Gatsby won't know what to render, if you manually navigate to `/blog` you will see your page. You will need `index.js` also because every website should have a `/` view. Sorry if I confused you about renaming the `index.js` file.
the `--prefix-path` is used only if you are not hosting your website at the root level for example when using Github pages

**Comment by KyleAMathews** [comment]
Yeah, you generally want to build all pages of your site with Gatsby but if you're using another tool for the homepage, you can either make all paths start with /blog or use path prefixing https://www.gatsbyjs.org/docs/path-prefix/

**Comment by AdvaithD** [comment]
@KyleAMathews Understood. I have the blog built with the '/blog' paths but I still am unable to have my static index.html and the assets(css, js) used for the landing page. Effectively, I want the bootstrap landing for the homepage, and '/blog' is where I want the blog to be hosted. 

Right now, the blog is works with the '/blog' path prefixes. However, the landing page isn't showing up as the landing, which was previously the case(landing showed u but gatsby blog didnt work)

Apologies for pestering you with these elementary issue but although I have a basic understanding of gatsby, haven't been able to figure this out.

**Comment by KyleAMathews** [comment]
If you want to have static pages not built with Gatsby, add them to the static folder

 https://www.gatsbyjs.org/docs/adding-images-fonts-files/#using-the-code-classlanguage-textstaticcode-folder

E.g. an index.html there will become your `/` page.

**Comment by AdvaithD** [comment]
@KyleAMathews I've tried multiple times and figured I'm doing something wrong.
the homepage(localhost:8000) gives the blog homepage instead of the static landing page, even after putting in the necessary files into /static
![](https://i.gyazo.com/c3f196266616766e91d1e3470b9af8b2.png)

However, when I click on the blog links, the have the desired /blog url's

I'd say that the only thing left is having the static bootstrap landing page at '/' and the blog homepage at '/blog'

**Comment by AdvaithD** [comment]
Also, would like to add that when I run `gatsby develop` the blog homepage shows up at '/' instead of the bootstrap landing page which is in the `/static` folder in the root of the project directory, although each blog post has `/blog` in its links.

**Comment by KyleAMathews** [comment]
Do you have a file at src/pages/index.js?

**Comment by AdvaithD** [comment]
@KyleAMathews yes
